src/main/java/org/hydev/mcpm/client/database/searchusecase/SearcherByKeyword.java

```diff
@@ -0,0 +1,77 @@
+package org.hydev.mcpm.client.database.searchusecase;
+
+import org.hydev.mcpm.client.models.PluginModel;
+
+import java.util.*;
+
+/**
+ * Searcher that returns a map based on keywords.
+ *
+ * @author Jerry Zhu (<a href="https://github.com/jerryzhu509">...</a>)
+ */
+public class SearcherByKeyword implements Searcher {
+
+    private static Map<String, List<PluginModel>> keywordMap = null;
+
+    /**
+     * Returns a dictionary mapping the different keywords to the matching plugins
+     * This function should ignore letter case in searching.
+     * For example, searching "java" would match "Java" as well.
+     * This function implements fuzzy search that doesn't require the exact phrase to be available,
+     * but requires all words in the phrase to be present.
+     * For example, searching "java 11" would match "java jdk 11" but not "java"
+     *
+     * @param plugins A list of all plugins in the database.
+     * @return A dictionary associating each keyword to the matching plugins.
+     */
+    @Override
+    public Map<String, List<PluginModel>> constructSearchMaps(List<PluginModel> plugins) {
+        Map<String, List<PluginModel>> models = new HashMap<>();
+        for (PluginModel plugin : plugins) {
+            // Get latest version
+            var v = plugin.getLatestPluginVersion();
+            if (v.isPresent() && v.get().meta() != null &&
+                    v.get().meta().description() != null && !v.get().meta().description().equals("")) {
+                String[] keywords = v.get().meta().description()
+                        .replaceAll("[^a-zA-Z ]", "").toLowerCase().split("\\s+");
```

Comment:
Nice regex!

Reply:
I didn’t see the carat so I thought it was deleting everything 💀
But isn’t the regex also deleting spaces? How do you split afterwards
Am I lacking critical information

Reply:
Honestly me too I have no idea what this matches to.

Reply:
^ means to exclude everything within the character class, in this case, every non (alphabetical character or space)

Reply:
Yeah so... the space is in the group right? We don't delete it :D

Reply:
What about newlines?

Reply:
Newlines would get deleted in the replaceAll... a bug?